### PR TITLE
ORC-227: [C++] Fix docker failure due to ExternalProject_Add and sign mismatch

### PR DESCRIPTION
**Pull request by majetideepak**

### CODE REVIEW

File: c++/src/ByteRLE.cc [hunk 1]
```diff
@@ -26,9 +26,9 @@
 
 namespace orc {
 
-  const size_t MINIMUM_REPEAT = 3;
-  const size_t MAXIMUM_REPEAT = 127 + MINIMUM_REPEAT;
-  const size_t MAX_LITERAL_SIZE = 128;
+  const int MINIMUM_REPEAT = 3;
+  const int MAXIMUM_REPEAT = 127 + MINIMUM_REPEAT;
+  const int MAX_LITERAL_SIZE = 128;
```
Comment (majetideepak) [inline c++/src/ByteRLE.cc]: @xndai @wgtmac These changes are required to avoid signed and unsigned mismatch warnings.

Comment (xndai) [inline c++/src/ByteRLE.cc]: I don't see the mismatch warnings on CentOS7 docker image. Where did you see that?

Comment (majetideepak) [inline c++/src/ByteRLE.cc]: I tried CentOS6 docker image. The warnings seem to be correct.
```
cc1plus: warnings being treated as errors
/root/orc/c++/src/ByteRLE.cc: In member function ‘void orc::ByteRleEncoderImpl::write(char)’:
/root/orc/c++/src/ByteRLE.cc:152: error: comparison between signed and unsigned integer expressions
/root/orc/c++/src/ByteRLE.cc:166: error: comparison between signed and unsigned integer expressions
/root/orc/c++/src/ByteRLE.cc:167: error: comparison between signed and unsigned integer expressions
/root/orc/c++/src/ByteRLE.cc:179: error: comparison between signed and unsigned integer expressions
make[2]: *** [c++/src/CMakeFiles/orc.dir/ByteRLE.cc.o] Error 1
make[1]: *** [c++/src/CMakeFiles/orc.dir/all] Error 2
```

Comment (xndai) [inline c++/src/ByteRLE.cc]: ok. Thanks for fixing this!

**Pull request closed without merging**